src/FakeItEasy/Core/ArgumentEqualityComparer.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace FakeItEasy.Core
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    internal class ArgumentEqualityComparer
+    {
+        private readonly IEnumerable<IArgumentEqualityComparer> argumentEqualityComparers;
+
+        public ArgumentEqualityComparer(IEnumerable<IArgumentEqualityComparer> argumentEqualityComparers)
+        {
+            this.argumentEqualityComparers = argumentEqualityComparers;
+        }
+
+        public bool AreEqual(object? expectedValue, object? argumentValue, Type parameterType)
+        {
+            var comparer = this.argumentEqualityComparers
+                .Where(c => c.CanCompare(parameterType))
+                .OrderByDescending(c => c.Priority)

```

Comment:
Sorting each time seems wasteful. Consider sorting when we construct the object. I fear this applies to our existing discoverable extension points as well.

Reply:
Good point. We do it for fake options builders and dummy factories. Not for argument value formatters, though.